### PR TITLE
Prefer normal stride selection logic when tensor is contiguous

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #143153

Fixes: https://github.com/pytorch/pytorch/issues/142024

### CODE REVIEW

**Comment by ezyang** [comment]
In your tlp, the contiguity is implicitly guarded via

```
+- LAMBDA_GUARD: L['aot6_sub_58'].stride()[0] == L['aot6_sub_58'].size()[1]  # (unknown source L['aot6_sub_58'].stride()[0], please file a bug)
```

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -3904,9 +3904,22 @@ def _create_symbolic_sizes_strides_storage_offset(
...
+        numel = math.prod(size)
+        if isinstance(numel, int) and numel < 2 or ex_stride[-1] != 1:
+            contiguous_striding = False
+        for i in range(len(size) - 2, -1, -1):
+            contiguous_stride = ex_stride[i + 1] * ex_size[i + 1]
+            if (
+                not isinstance(contiguous_stride, sympy.Symbol)
+                and ex_stride[i] != contiguous_stride
+            ):
+                contiguous_striding = False
```
Comment (aorenste) [inline torch/fx/experimental/symbolic_shapes.py:3917]: nit: you could break after this, right?

File: test/dynamo/test_repros.py [hunk 2]
```diff
@@ -5812,6 +5812,23 @@ def fn(x, y):
...
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(aot6_sub_58, aot6_mul_170):
+            aot6_unsqueeze_14 = torch.ops.aten.unsqueeze.default(aot6_mul_170, 1)
+            return torch.ops.aten.mul.Tensor(aot6_sub_58, aot6_unsqueeze_14)
+
+        aot6_sub_58 = torch.randn(2, 1)
+        torch._dynamo.decorators.mark_unbacked(aot6_sub_58, 1)
+        aot6_mul_170 = torch.randn(2)
+
+        fn(aot6_sub_58, aot6_mul_170)
```
[the remark this answers is not in the document]
Comment (bobrenjc93) [reply test/dynamo/test_repros.py:5830]: yep it used to crash, i'll add a comment

**Comment by bobrenjc93** [comment]
Attempt 3: https://github.com/pytorch/pytorch/pull/143365

**Pull request closed without merging**